ui/src/app/kvm/components/KVMResourcesCard/KVMResourcesCard.tsx

```diff
@@ -44,12 +45,28 @@ const KVMResourcesCard = ({
           <h5 className="p-text--paragraph u-flex--between u-no-max-width">
             <span data-test="kvm-resources-card-title">{title}</span>
             <ContextualMenu
-              dropdownContent={<></>}
+              dropdownClassName="kvm-machine-list-modal"
+              dropdownContent={
+                <MachineListTable
+                  hiddenColumns={[
+                    "power",
+                    "owner",
+                    "pool",
+                    "zone",
+                    "fabric",
+                    "disks",
+                    "storage",
+                  ]}
+                  showActions={false}
+                  paginateLimit={5}
+                ></MachineListTable>
+              }
               hasToggleIcon
               toggleAppearance="base"
               toggleClassName="kvm-resources-card__vms-button is-dense"
               toggleDisabled={vms.length === 0}
               toggleLabel={pluralize("machine", vms.length, true)}
+              position="left"
```

Comment:
It would be good if the ContextualMenus has an `auto` position. As it stands currently, if the page is resized to be small, or its the 4th of 4 NUMA nodes the dropdown extends past the edge of the screen.